File: 2025-10-October-GFG-POTD/October 09 - Postorder Traversal/Solution.py
# Definition for a binary tree node
class Node:
    def __init__(self, val):
        self.data = val
        self.left = None
        self.right = None

class Solution:
    def postOrder(self, root):
        result = []
        self._postOrderHelper(root, result)
        return result

    def _postOrderHelper(self, node, result):
        if node is None:
            return
        
        # 1️⃣ Visit left subtree
        self._postOrderHelper(node.left, result)
        # 2️⃣ Visit right subtree
        self._postOrderHelper(node.right, result)
        # 3️⃣ Visit node itself
        result.append(node.data)

File: 2025-10-October-GFG-POTD/October 09 - Postorder Traversal/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

struct Node {
    int data;
    Node* left;
    Node* right;
    Node(int val) {
        data = val;
        left = right = nullptr;
    }
};

class Solution {
public:
    vector<int> postOrder(Node* root) {
        vector<int> result;
        postOrderHelper(root, result);
        return result;
    }

private:
    void postOrderHelper(Node* node, vector<int>& result) {
        if (node == nullptr) return;

        // 1️⃣ Visit left subtree
        postOrderHelper(node->left, result);
        // 2️⃣ Visit right subtree
        postOrderHelper(node->right, result);
        // 3️⃣ Visit node itself
        result.push_back(node->data);
    }
};
